src/shapes/Text/StyledText.ts

```diff
@@ -179,21 +181,18 @@ export abstract class StyledText<
     }
   }
 
-  private _extendStyles(index: number, styles: TextStyleDeclaration) {
+  private _extendStyles(index: number, styles: TextStyleDeclaration): void {
     const { lineIndex, charIndex } = this.get2DCursorLocation(index);
 
     if (!this._getLineStyle(lineIndex)) {
       this._setLineStyle(lineIndex);
     }
 
-    if (!this._getStyleDeclaration(lineIndex, charIndex)) {
+    if (!Object.keys(this._getStyleDeclaration(lineIndex, charIndex)).length) {
```

Comment:
this is the only code change i made.
there were 5 occurrencies in which returning undefined from `_getStyleDeclaration` needed to be fixed with an `|| {}` while just one where returning undefined ( or null ) was useful, to know if there was no style.

Reply:
I guess this is fine but BREAKING if anyone used it before

Reply:
i would make this breaking, and i someone asks, i would add the hasStyleDeclaration method that return true/false, or maybe i can add it right away

Reply:
I think you should
Then there is a clear migration path (Object.keys is pretty clear also but...)